### PR TITLE
BREAKING CHANGE: Refactor enum identifiers to TitleCase and apply some changes in AST nodes 

### PR DESCRIPTION
**Pull request by MaxGraey**
Changes:

+ Canonicalize all enum identifiers from mixed (`UPPER`, `UPPER_SNAKE` and `TitleCase`) case to `TitleCase` only
+ Rename `statement` to `body` in all loop-related AST nodes  in `ast.ts`
+ Rename `statements` to `bodyStatements` for TryStatement node  in `ast.ts`
+ Add `Node > .isEmpty` in `ast.ts`
+ Add `SwitchCase > .isDefault`  in `ast.ts`

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ export const enum FeatureFlags {
   Atomics = 1 /* _BinaryenFeatureAtomics */,
   MutableGlobals = 2 /* _BinaryenFeatureMutableGlobals */,
   TruncSat = 4 /* _BinaryenFeatureNontrappingFPToInt */,
-  SIMD = 8 /* _BinaryenFeatureSIMD128 */,
+  Simd = 8 /* _BinaryenFeatureSIMD128 */,
```
Comment (dcodeIO) [inline src/module.ts:175]: Similar to other abbreviations like `GC`. Should probably remain `SIMD`.

File: src/module.ts [hunk 2]
```diff
@@ -181,8 +181,7 @@ export const enum FeatureFlags {
   MultiValue = 512 /* _BinaryenFeatureMultivalue */,
   GC = 1024 /* _BinaryenFeatureGC */,
   Memory64 = 2048 /* _BinaryenFeatureMemory64 */,
-  // GCNNLocals is for off-spec experimentation
-  RelaxedSIMD = 8192 /* _BinaryenFeatureRelaxedSIMD */,
+  RelaxedSimd = 8192 /* _BinaryenFeatureRelaxedSIMD */,
```
Comment (dcodeIO) [inline src/module.ts:184]: Similar to `SIMD`: `RelaxedSIMD`

File: src/parser.ts [hunk 3]
```diff
@@ -3912,54 +3912,54 @@ export class Parser extends DiagnosticEmitter {
...
+      case Token.Identifier: {
         let identifierText = tn.readIdentifier();
         if (identifierText == "null") return Node.createNullExpression(tn.range()); // special
         let identifier = Node.createIdentifierExpression(identifierText, tn.range(startPos, tn.pos));
-        if (tn.skip(Token.TEMPLATELITERAL)) {
+        if (tn.skip(Token.TemplateLiteral)) {
           return this.parseTemplateLiteral(tn, identifier);
         }
-        if (tn.peek(true) == Token.EQUALS_GREATERTHAN && !tn.nextTokenOnNewLine) {
+        if (tn.peek(true) == Token.EqualsGreaterThan && !tn.nextTokenOnNewLine) {
```
Comment (dcodeIO) [inline src/parser.ts:3941]: `EqualsGreaterThan`, and similar combined tokens, are a little strange. In `EQUALS_GREATERTHAN`, it was clear that this is `=` + `>`. Would it perhaps make sense to use `Greaterthan`, as if it was a single word?

File: src/parser.ts [hunk 4]
```diff
@@ -4140,7 +4140,7 @@ export class Parser extends DiagnosticEmitter {
           break;
         }
         // InstanceOfExpression
-        case Token.INSTANCEOF: {
+        case Token.InstanceOf: {
```
Comment (dcodeIO) [inline src/parser.ts:4143]: This one is somewhat similar, in that the operation here is `instanceof`, so perhaps `Instanceof` to indicate?

Comment (dcodeIO) [reply src/parser.ts:4143]: Otoh, I see now that this notation becomes weird on its own on `OpenBracket` and such. Hmm.

File: src/module.ts (hunk 1 above)
Comment (MaxGraey) [reply src/module.ts]: Done!

File: src/parser.ts (hunk 4 above)
Comment (MaxGraey) [reply src/parser.ts]: I switched to `Instanceof`

File: src/parser.ts (hunk 3 above)
Comment (MaxGraey) [reply src/parser.ts:3941]: `Greaterthan` is really hard for readability compare to `GreaterThan`. Especially for compound variants. Just compare:
```ts
GREATERTHAN_GREATERTHAN_GREATERTHAN_EQUALS
GreaterthanGreaterthanGreaterthanEquals
GreaterThanGreaterThanGreaterThanEquals
```

**Review by dcodeIO — APPROVED** [review]

**Pull request merged**